src/libPMacc/include/pmacc_types.hpp

```diff
@@ -176,16 +178,14 @@ enum EventType
  * you must align all array and structs which can used on device
  * @param byte byte of data which must aligned
  */
-#define __optimal_align__(byte)   \
-        __align__(                \
-        ((byte)==1?1:             \
-        ((byte)<=2?2:             \
-        ((byte)<=4?4:             \
-        ((byte)<=8?8:             \
-        ((byte)<=16?16:           \
-        ((byte)<=32?32:           \
-        ((byte)<=64?64:128        \
-        ))))))))
+#define __optimal_align__(byte)                                                \
+    __align__(                                                                 \
+        PMACC_MIN(                                                             \
+            /* 32 byte is the L2 cache line size of NVIDIA GPUs */             \
+            PMACC_MIN(PMACC_ROUND_UP_NEXT_POW2(byte),32),                      \
```

Comment:
This will not work. 32 byte is the maximum alignment possible that is compatible between nvcc and gcc 4.3+.
So there is no change to the current, buggy behaviour for structs >32byte.
Also it probably would be better, to state, that this is the maximum vector load size (which happens to be the cache line size)